test/mailers/previews/user_mailer_preview.rb

```diff
@@ -199,4 +214,58 @@ def anonymous_or_unrevealed_data(status)
     item = create(:work, authors: [user.default_pseud], collections: [collection])
     [user, collection, item]
   end
+
+  def recipient_notification_data(count)
+    fandoms = []
+    relationships = []
+    characters = []
+    tags = []
+    categories = [
+      ArchiveConfig.CATEGORY_GEN_TAG_NAME,
+      ArchiveConfig.CATEGORY_SLASH_TAG_NAME,
+      ArchiveConfig.CATEGORY_HET_TAG_NAME,
+      ArchiveConfig.CATEGORY_FEMSLASH_TAG_NAME,
+      ArchiveConfig.CATEGORY_MULTI_TAG_NAME,
+      ArchiveConfig.CATEGORY_OTHER_TAG_NAME
+    ]
+    warnings = [
+      ArchiveConfig.WARNING_DEFAULT_TAG_NAME,
+      ArchiveConfig.WARNING_NONE_TAG_NAME,
+      ArchiveConfig.WARNING_VIOLENCE_TAG_NAME,
+      ArchiveConfig.WARNING_DEATH_TAG_NAME,
+      ArchiveConfig.WARNING_NONCON_TAG_NAME
+    ]
+    series_list = []
+    
+    count = 1 if count < 1
+    if count >= 1
+      (1..count).each do |n| 
+        fandoms.append("fandom_#{n}")
+        relationships.append("relationship_#{n}")
+        characters.append("character_#{n}")
+        tags.append("tag_#{n}")
+        series_list.append(create(:series))
+      end
+      warnings = warnings[0...count] if count <= 5
```

Comment:
This is much simpler done with `ArchiveWarning.first(count).pluck(:name)` instead of the hardcoded list. It even behaves correctly when count is too large.